src/app/shared/header/header.component.ts

```diff
@@ -1,81 +1,42 @@
-import {
-  animate,
-  state,
-  style,
-  transition,
-  trigger
-} from '@angular/animations';
-import { AfterViewInit, Component, HostBinding } from '@angular/core';
-import { fromEvent } from 'rxjs';
-import {
-  distinctUntilChanged,
-  filter,
-  map,
-  pairwise,
-  share,
-  throttleTime
-} from 'rxjs/operators';
-
-enum VisibilityState {
-  Visible = 'visible',
-  Hidden = 'hidden'
-}
-
-enum Direction {
-  Up = 'Up',
-  Down = 'Down'
-}
+import { AfterViewInit, Component } from '@angular/core';
+import { fromEvent, interval } from 'rxjs';
+import { debounce } from 'rxjs/operators';
 
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
-  styleUrls: ['./header.component.scss'],
-  animations: [
-    trigger('toggle', [
-      state(
-        VisibilityState.Hidden,
-        style({opacity: 0, transform: 'translateY(-100%)'})
-      ),
-      state(
-        VisibilityState.Visible,
-        style({opacity: 1, transform: 'translateY(0)'})
-      ),
-      transition('* => *', animate('200ms ease-in'))
-    ])
-  ]
+  styleUrls: ['./header.component.scss']
 })
 export class HeaderComponent implements AfterViewInit {
-  private isVisible = true;
 
-  @HostBinding('@toggle')
-  get toggle(): VisibilityState {
-    return this.isVisible ? VisibilityState.Visible : VisibilityState.Hidden;
-  }
-
-  /**
-   * make reactive sticky header
-   * swipe up - hide
-   * swipe down - show
-   */
   ngAfterViewInit() {
-    const scroll$ = fromEvent(window, 'scroll').pipe(
-      throttleTime(10),
-      map(() => window.pageYOffset),
-      pairwise(),
-      map(([y1, y2]): Direction => (y2 < y1 ? Direction.Up : Direction.Down)),
-      distinctUntilChanged(),
-      share()
-    );
-
-    const goingUp$ = scroll$.pipe(
-      filter(direction => direction === Direction.Up)
-    );
-
-    const goingDown$ = scroll$.pipe(
-      filter(direction => direction === Direction.Down)
-    );
-
-    goingUp$.subscribe(() => (this.isVisible = true));
-    goingDown$.subscribe(() => (this.isVisible = false));
+    const header = document.getElementById('header');
+
+    const isScrolling = fromEvent(window, 'scroll')
+      .subscribe(() => {
+        let doc = document.documentElement;
+        doc = (doc.clientHeight) ? doc : document.body;
+
+        if (doc.scrollTop >= 0 && doc.scrollTop <= 50) {
+          header.classList.add('notransition');
+          header.classList.remove('hide'); // add header without delay when user scroll to top of the page
+        } else {
+          header.classList.add('hide');
+          header.classList.remove('notransition'); // run when user is scrolling
+        }
+      });
+
+    const stoppedScrolling = fromEvent(window, 'scroll').pipe(
+      debounce(() => interval(2000))
+    ).subscribe(() => {
+      header.classList.remove('hide');
```

Comment:
we have such directive as ngClass, please use it

Reply:
Done.